message_app_server/message_app/send_message.py

```diff
@@ -97,25 +96,15 @@ def get_ten_messages(conversation_id, cursor=None):
 # ================== Sockets for Conversations Container ==========================
 
 # Joinning a conversation
-@socketio.on("join", namespace="/messages")
-def joining(data):
-    username = data["username"]
-    conversation_id = data["conversation_id"]
-
-    join_room(conversation_id)
+@socketio.on("join_conversation", namespace="/messages")
+def join_conversation(data):
+    old_conversation_id = data["oldConversationID"]
+    new_conversation_id = data["newConversationID"]
 
-    # Send a message that inform new user has join the conversation
-    emit("message_handler_client", {"username": username, "conversation_id": conversation_id, "join": True}, room=conversation_id)
-
-# Leaving a conversation
-@socketio.on("leave", namespace="/messages")
-def leaving(data):
-    username = data["username"]
-    conversation_id = data["conversation_id"]
-    leave_room(conversation_id)
+    if (old_conversation_id):
+        leave_room(old_conversation_id)
 
-    # Send a message that inform an user has left the conversation
-    emit("message_handler_client", {"username": username, "conversation_id": conversation_id, "join": False}, room=conversation_id)
+    join_room(new_conversation_id)
```

Comment:
So the logic here is that, if user wants to join a new conversation, we will disconnect user from the current conversation (if any)?

Reply:
Yes